File: src/dataset/cpp/grains_exercism-felbit/gemini/GeminiFlash_grains_v2.cpp
#include "grains.h"
#include <cmath>

namespace grains
{

    ull square(unsigned short _square)
    {
        return 1ULL << (_square - 1);
    }

    ull total()
    {
        return (1ULL << 64) - 1;
    }

}

File: src/dataset/typescript/grains_exercism-wackerow/gemini/GeminiFlash_grains_v2.ts
export default class Grains {
  private static readonly TWO_POW_64 = BigInt("18446744073709551615");

  public static square(digit: number): bigint {
    if (digit <= 0 || digit > 64 || digit % 1 !== 0) {
      throw 'Chessboard location must be integer from 1 to 64.';
    }
    return BigInt(1) << BigInt(digit - 1);
  }

  public static total(): bigint {
    return Grains.TWO_POW_64;
  }
}